### PR TITLE
Update gibct-data-service Rails to version 5.0.7.2 #19605

### PR DESCRIPTION
**Pull request by zurbergram**
## Description 
Update gibct-data-service Rails to version 5.0.7.2 #19605
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19605

Some changes are based off what was done in https://github.com/department-of-veterans-affairs/vets-api/pull/2833

https://hashrocket.com/blog/posts/how-to-upgrade-to-rails-5 

## Testing
- [ ] Smoke test GIDS application and usage of GIDS endpoints

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by sonntag-adam — APPROVED** [review]

File: app/models/upload.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def self.last_uploads
   end
 
   def self.last_uploads_rows
-    uploads = Upload.last_uploads
+    uploads = Upload.last_uploads.to_a
```
Comment (kfrz) [inline app/models/upload.rb]: What's the warrant for this change being part of the `rails` upgrade?

File: bin/setup [hunk 2]
```diff
@@ -1,29 +1,36 @@
 #!/usr/bin/env ruby
+# frozen_string_literal: true
```
Comment (kfrz) [inline bin/setup]: This isn't strictly necessary, and rubocop will look for it on the first line of the file anyhow... so this can be removed

File: bin/setup [hunk 3]
```diff
@@ -1,29 +1,36 @@
...
+
 require 'pathname'
+require 'fileutils'
+include FileUtils # rubocop:disable Style/MixinUsage
```
Comment (kfrz) [inline bin/setup]: Why is this here?

File: bin/update [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'pathname'
+require 'fileutils'
+include FileUtils # rubocop:disable Style/MixinUsage
```
Comment (kfrz) [inline bin/update]: Why is this here?

File: spec/rails_helper.rb [hunk 5]
```diff
@@ -61,6 +61,7 @@
   # database_cleaner configuration
   # Clear the entire DB before tests begin
   config.before(:suite) do
+    DatabaseCleaner.allow_remote_database_url = true
```
Comment (kfrz) [inline spec/rails_helper.rb]: Can you add an explanation as to why we need to allow remote db url for our tests?

**Review by kfrz — CHANGES_REQUESTED** [review]
Overall it looks like this is close to merge-ready. 

With very large PRs such as this, it's often much easier for a developer to view individual commits, rather than scrolling through hundreds of diffs on a single page. 

Unfortunately the commits tab on this PR is mostly full of non-informative comments which makes that process more difficult. In the future I'd love to see more atomic, descriptive, [imperative-mode](https://chris.beams.io/posts/git-commit/) messages, _especially_ on large efforts like this.

Other than a few minor questions/changes, I think this is in good shape.

File: bin/setup (hunk 3 above)
Comment (zurbergram) [reply bin/setup]: The rubocop disable?

File: app/models/upload.rb (hunk 1 above)
Comment (zurbergram) [reply app/models/upload.rb]: Removed

File: bin/setup (hunk 2 above)
Comment (zurbergram) [reply bin/setup]: Removed

File: bin/setup (hunk 3 above)
Comment (zurbergram) [reply bin/setup]: Removed the rubocop disable

File: bin/update (hunk 4 above)
Comment (zurbergram) [reply bin/update]: Removed

File: spec/rails_helper.rb (hunk 5 above)
Comment (zurbergram) [reply spec/rails_helper.rb]: in http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fgibct-data-service/detail/rails_5.0.7.2/18/pipeline as well as few other builds received the error below

```
An error occurred in a `before(:suite)` hook.
Failure/Error: DatabaseCleaner.clean_with(:truncation)

DatabaseCleaner::Safeguard::Error::RemoteDatabaseUrl:
  ENV['DATABASE_URL'] is set to a remote URL. Please refer to https://github.com/DatabaseCleaner/database_cleaner#safeguards
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/safeguard.rb:21:in `run'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/safeguard.rb:69:in `block in run'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/safeguard.rb:69:in `each'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/safeguard.rb:69:in `run'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/base.rb:19:in `initialize'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:46:in `new'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:46:in `connections'
# /usr/local/bundle/gems/database_cleaner-1.7.0/lib/database_cleaner/configuration.rb:91:in `clean_with'
# ./spec/rails_helper.rb:64:in `block (2 levels) in <top (required)>'
```

File: app/models/upload.rb (hunk 1 above)
Comment (zurbergram) [reply app/models/upload.rb]: Error from http://jenkins.vfs.va.gov/blue/organizations/jenkins/testing%2Fgibct-data-service/detail/rails_5.0.7.2/22/pipeline was reason I originally added it

Comment (zurbergram) [reply app/models/upload.rb]: `Upload.last_uploads` returns a `Upload::ActiveRecord_Relation` object in test `spec/controllers/dashboards_controller_spec.rb`

Comment (zurbergram) [reply app/models/upload.rb]: I think there was some kind of implicit conversion from `Upload::ActiveRecord_Relation` to array or something because in master branch the test passes fine

I did a `binding.pry()` in `Upload::last_uploads_rows` to check class of uploads object, it shows that `uploads` is a `Upload::ActiveRecord_Relation` and then let me `uploads.push(Upload.new)` without any issue

Comment (zurbergram) [reply app/models/upload.rb]: There is a discussion in https://github.com/rails/rails/issues/25906 that talks about how ActiveRecord_Relations are no longer mutable and so the ability to change them was removed in Rails 5

File: spec/rails_helper.rb (hunk 5 above)
Comment (kfrz) [reply spec/rails_helper.rb]: Could we do something like:

```suggestion
    DatabaseCleaner.allow_remote_database_url = ENV['CI']
```
This should safeguard against accidentally connecting to remote db in development, to avoid a footgun

File: bin/setup [hunk 6]
```diff
@@ -1,29 +1,35 @@
 #!/usr/bin/env ruby
+
 require 'pathname'
+require 'fileutils'
+include FileUtils
```
Comment (kfrz) [inline bin/setup]: Is it necessary to include *and* require here

File: bin/setup [hunk 7]
```diff
@@ -1,29 +1,35 @@
...
 
 # path to your application root.
-APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
+APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
```
Comment (kfrz) [inline bin/setup]: Prefer `__dir__` method over `__FILE__` constant

Comment (kfrz) [reply bin/setup]: Also applies to other bin/scripts

File: bin/setup (hunk 6 above)
Comment (zurbergram) [reply bin/setup]: removed include

File: bin/setup (hunk 7 above)
Comment (zurbergram) [reply bin/setup]: switched all bin/scripts to `__dir__`

File: Gemfile [hunk 8]
```diff
@@ -24,11 +24,7 @@ gem 'cancancan', '~> 1.13', '>= 1.13.1'
 gem 'govdelivery-tms', '2.8.4', require: 'govdelivery/tms/mail/delivery_method'
 # Use postgresql as the database for Active Record
 gem 'pg', '~> 0.15'
-
-# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rubyzip', '>= 1.3.0'
-
-gem 'sentry-raven', '~> 2.3.0'
+gem 'sentry-raven', '2.9.0' # don't change gem version unless sentry server is also upgraded
```
Comment (annaswims) [inline Gemfile]: I don't think there's a reason to updade sentry at the same time as rails.  Please do it before or after.

Comment (zurbergram) [reply Gemfile]: This was part of the "it's in vets-api" changes, reverted

**Review by kfrz — APPROVED** [review]
